indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -178,14 +183,24 @@ public final TaskStatus run(TaskToolbox taskToolbox) throws Exception
       throw e;
     }
     finally {
-      cleanUp(taskToolbox, taskStatus);
+      try {
+        cleanUp(taskToolbox, taskStatus);
+      }
+      finally {
+        cleanupCompletionLatch.countDown();
+      }
     }
   }
 
   public abstract TaskStatus runTask(TaskToolbox taskToolbox) throws Exception;
 
-  public void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
+  protected void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
   {
+    if (Thread.currentThread().isInterrupted()) {
+      // clears the interrupted status so the subsequent cleanup work can continue without interruption
+      Thread.interrupted();
```

Comment:
I'm not exact sure whether this has a bigger impact on somewhere else. fyi @kfaraz